Allow audit URLs to be used instead of diffs

adam-ja opened the pull request:
Sometimes, we need to release quick fixes that were not reviewed. We still want to add these to the release app for visibility and, rather than linking to nothing, we can link to the audit on phab.

For example, link to http://phab.dev.ebuyer.com/rGENONEfcc395703a1f8edee04db518ed56c121705900fe rather than http://phab.dev.ebuyer.com/D5610

This patch updates the Add Diff form to indicate this possibility, and improves the styling of the release view so that lengthy commit hashes do not overflow other elements of a release (such as the Release button/Released status text).

Commit:
Fix release cut off check

Previously the cut off warning would always appear if the current hour
was 14 or higher. This meant that a release two or more days into the
future could show the warning prematurely (e.g. we should be able to add
tickets to a release on Wednesday after 2pm on Monday but the warning
was still showing).

Now, the cut off warning actually takes into account the date of the
release and only shows the warning after 2pm on the last working day
before the release.

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Allow audit URLs to be used instead of diffs

Sometimes, we need to release quick fixes that were not reviewed. We
still want to add these to the release app for visibility and, rather
than linking to nothing, we can link to the audit on phab.

For example, link to
http://phab.dev.ebuyer.com/rGENONEfcc395703a1f8edee04db518ed56c121705900fe
rather than http://phab.dev.ebuyer.com/D5610

This patch updates the Add Diff form to indicate this possibility, and
improves the styling of the release view so that lengthy commit hashes
do not overflow other elements of a release (such as the Release
button/Released status text).

adam-ja commented:
BEFORE:
![image](https://user-images.githubusercontent.com/3234173/32778179-ce08300c-c930-11e7-8480-a17fe67dd34e.png)
![image](https://user-images.githubusercontent.com/3234173/32778189-d561f82e-c930-11e7-8ebe-684682d8de26.png)


AFTER:
![image](https://user-images.githubusercontent.com/3234173/32778145-abaa04b8-c930-11e7-827c-4c3b046c494b.png)
![image](https://user-images.githubusercontent.com/3234173/32778154-b9eee1ba-c930-11e7-9b55-b1df2de7e877.png)

tkjn commented on views/partials/release-ticket-diff-add.jade:
```diff
@@ -9,12 +9,12 @@ div(class='modal-body')
   )
     div(class='form-group')
-      label(class='control-label col-sm-2') Phabricator Diff
+      label(class='control-label col-sm-2') Phabricator Diff/Audit
```
I don't think it's necessarily a link to an "Audit". It's a link to a "Commit", sometimes those commits require Audit.
(shrug)

alancrisp replied:
That is indeed the case and, depending on which repo is being committed to, the audit may not be created immediately. "Commit" would be more appropriate terminology.

adam-ja replied:
Done.

Commit:
Improve terminology from Audit to Commit

We can link to any commit on phabricator, regardless of whether or not
that commit requires auditing.
